pkg/providers/tinkerbell/comparator_test.go

```diff
@@ -0,0 +1,91 @@
+package tinkerbell_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/providers/tinkerbell"
+	"github.com/aws/eks-anywhere/pkg/utils/ptr"
+)
+
+func TestControlPlaneUnchanged(t *testing.T) {
+	g := NewWithT(t)
+
+	oldKubeadmControlPlane := comparatorKubeadmControlPlane()
+	newKubeadmControlPlane := comparatorKubeadmControlPlane()
+
+	g.Expect(tinkerbell.ControlPlaneVersionChange(oldKubeadmControlPlane, newKubeadmControlPlane), false)
+	g.Expect(tinkerbell.ControlPlaneReplicasChange(oldKubeadmControlPlane, newKubeadmControlPlane), false)
+}
+
+func TestControlPlaneChanged(t *testing.T) {
+	g := NewWithT(t)
+
+	oldKubeadmControlPlane := comparatorKubeadmControlPlane()
+	newKubeadmControlPlane := comparatorKubeadmControlPlane()
+	newKubeadmControlPlane.Spec.Version = "1.23"
+	newKubeadmControlPlane.Spec.Replicas = ptr.Int32(2)
+
+	g.Expect(tinkerbell.ControlPlaneVersionChange(oldKubeadmControlPlane, newKubeadmControlPlane), true)
+	g.Expect(tinkerbell.ControlPlaneReplicasChange(oldKubeadmControlPlane, newKubeadmControlPlane), true)
```

Comment:
could we have independent test per tested method?

Reply:
Yeah I'll split those to be per method.